wavefront/server/modules/agents_module/agents_module/services/workflow_crud_service.py

```
@@ -548,60 +640,105 @@ async def update_workflow(
+                    f'Workflow {workflow_id} has no version {target_version} to branch from'
+                )
+            write_version = max(v.version for v in existing_versions) + 1
+            try:
+                await self.workflow_version_repository.create(
+                    workflow_id=workflow.id, version=write_version, is_deleted=False
+                )
+            except IntegrityError:
+                # Another concurrent create_new_version call already took write_version.
+                raise ValueError(
+                    f'Concurrent version creation conflict for workflow {workflow_id} - please retry'
+                )
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)